jupyter-images/gpu/Dockerfile

```diff
@@ -15,19 +15,24 @@ RUN apt-get update --yes && \
     apt-get upgrade --yes && \
     apt-get install --yes --no-install-recommends \
     bzip2  ca-certificates locales sudo wget software-properties-common \
-    libproj-dev proj-data proj-bin libgeos-dev && \
+    libproj-dev proj-data proj-bin libgeos-dev libx11-6 libgl1-mesa-glx \
+    libxrender1 xvfb && \
     # updating Python messes up Tensorflow from base container, unfort.
     # add-apt-repository ppa:deadsnakes/ppa && apt-get update --yes && \
     # apt-get install -y python3.10 python3.10-distutils && \
     # curl -sS https://bootstrap.pypa.io/get-pip.py | python3.10 && \
     # ln -sfn /usr/bin/python3.10 /usr/bin/python3 && \
     # ln -sfn /usr/bin/python3 /usr/bin/python && \
     # ln -sfn /usr/bin/pip3 /usr/bin/pip && \
-    pip install --no-cache-dir jupyterhub==3.0.0 && \
-    pip install --no-cache-dir --upgrade jupyterlab notebook jupyter_server && \
-    python3 -m pip install cartopy catboost metpy minisom netCDF4 pillow \
-    pyvista seaborn shapely torch torchaudio torchvision verde xarray \
-    --extra-index-url https://download.pytorch.org/whl/cu112 && \
+    python3 -m pip uninstall vtk -y && \
+    python3 -m pip install --no-cache-dir jupyterhub==3.0.0 jupyterlab>=3 \
+    notebook jupyter_server trame cartopy catboost metpy minisom netCDF4 \
+    pillow pyvista[all,trame] pyvista-xarray seaborn shapely torch torchaudio \
+    torchvision verde xarray ipywidgets jupyterlab_widgets jupyter-server-proxy \
+    --upgrade && \
+    python3 -m pip install --no-cache-dir --upgrade --extra-index-url \
+    https://wheels.vtk.org vtk-osmesa --extra-index-url \
```

Comment:
Thanks @banesullivan. So in that snippet, if I just call `python3 -m pip uninstall vtk -y` last, instead of first. We are good?

Reply:
Also, you'll notice that `trame` is referred to twice. Is that necessary?

Reply:
> you'll notice that trame is referred to twice. Is that necessary?

Not necessary. Keep the `trame` that is in `pyvista`'s extra options. i.e. `pyvista[all,trame]`

> So in that snippet, if I just call python3 -m pip uninstall vtk -y last, instead of first. We are good?

We want it right before the installation of `vtk-osmesa` and for that installation to be last.

```
python3 -m pip uninstall vtk -y && \
python3 -m pip install --no-cache-dir --upgrade --extra-index-url \
https://wheels.vtk.org vtk-osmesa --extra-index-url \
```